File: brownfield-backend/src/main/java/com/petclinic/vet/VetService.java
package com.petclinic.vet;

import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class VetService {
    
    private final VetRepository vetRepository;
    
    public VetService(VetRepository vetRepository) {
        this.vetRepository = vetRepository;
    }
    
    public List<Vet> findAll() {
        return vetRepository.findAll();
    }
    
    public Vet findById(Long id) {
        return vetRepository.findById(id).orElse(null);
    }
    
    public Vet findByName(String name) {
        return vetRepository.findByName(name);
    }
}

File: brownfield-backend/src/vet/vetService.ts
import { Vet } from "./vet";
import { VetRepository } from "./vetRepository";

export class VetService {
  private vetRepository: VetRepository;

  constructor(vetRepository?: VetRepository) {
    this.vetRepository = vetRepository ?? new VetRepository();
  }

  findAll(): Promise<Vet[]> {
    return this.vetRepository.findAll();
  }

  findById(id: number): Promise<Vet | null> {
    return this.vetRepository.findById(id);
  }

  findByName(name: string): Promise<Vet | null> {
    return this.vetRepository.findByName(name);
  }
}
